src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -14,223 +14,49 @@
- *   - one adaptation set contains a single media type
- *   - The order of verifications are:
- *       1. mimeType
- *       2. Role
- *       3. codec
- *
- * Note: This is based on DASH-IF-IOP-v4.0 with some more freedom.
- * @param {Object} adaptation
- * @returns {string} - "audio"|"video"|"text"|"image"|"metadata"|"unknown"
- */
-function inferAdaptationType(
-  adaptationMimeType : string|null,
-  representationMimeTypes : string[],
-  adaptationCodecs : string|null,
-  representationCodecs : string[],
-  adaptationRoles : IScheme[]|null
-) : string {
-
-  function fromMimeType(
-    mimeType : string,
-    roles : IScheme[]|null
-  ) : string|undefined {
-    const topLevel = mimeType.split("/")[0];
-    if (arrayIncludes(KNOWN_ADAPTATION_TYPES, topLevel)) {
-      return topLevel;
-    }
-
-    if (mimeType === "application/bif") {
-      return "image";
-    }
-
-    if (mimeType === "application/ttml+xml") {
-      return "text";
-    }
-
-    // manage DASH-IF mp4-embedded subtitles and metadata
-    if (mimeType === "application/mp4") {
-      if (roles != null) {
-        if (
-          arrayFind(roles, (role) =>
-            role.schemeIdUri === "urn:mpeg:dash:role:2011" &&
-            arrayIncludes(SUPPORTED_TEXT_TYPES, role.value)
-          ) != null
-        ) {
-          return "text";
-        }
-      }
-      return "metadata";
-    }
-  }
-
-  function fromCodecs(codecs : string) {
-    switch (codecs.substr(0, 3)) {
-      case "avc":
-      case "hev":
-      case "hvc":
-      case "vp8":
-      case "vp9":
-      case "av1":
-        return "video";
-      case "vtt":
-        return "text";
-      case "bif":
-        return "image";
-    }
-
-    switch (codecs.substr(0, 4)) {
-      case "mp4a":
-        return "audio";
-      case "wvtt":
-      case "stpp":
-        return "text";
-    }
-  }
-
-  if (adaptationMimeType != null) {
-    const typeFromMimeType = fromMimeType(adaptationMimeType, adaptationRoles);
-    if (typeFromMimeType != null) {
-      return typeFromMimeType;
-    }
-  }
-
-  if (adaptationCodecs != null) {
-    const typeFromCodecs = fromCodecs(adaptationCodecs);
-    if (typeFromCodecs != null) {
-      return typeFromCodecs;
-    }
-  }
-
-  for (let i = 0; i < representationMimeTypes.length; i++) {
-    const representationMimeType = representationMimeTypes[i];
-    if (representationMimeType != null) {
-      const typeFromMimeType = fromMimeType(representationMimeType, adaptationRoles);
-      if (typeFromMimeType != null) {
-        return typeFromMimeType;
-      }
-    }
-  }
-
-  for (let i = 0; i < representationCodecs.length; i++) {
-    const codecs = representationCodecs[i];
-    if (codecs != null) {
-      const typeFromMimeType = fromCodecs(codecs);
-      if (typeFromMimeType != null) {
-        return typeFromMimeType;
-      }
-    }
-  }
-
-  return "unknown";
+import { IParsedDASHPeriod } from "./Period";
+
+export interface IParsedDASHManifest {
```

Comment:
To check if an ``extend`` with a redefined ``periods`` work if this is the only difference.